README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# okcupiddata (revised in 2021)

[![CRAN\_Status\_Badge](http://www.r-pkg.org/badges/version/okcupiddata)](http://cran.r-project.org/package=okcupiddata)
[![CRAN RStudio mirror
downloads](http://cranlogs.r-pkg.org/badges/okcupiddata)](http://www.r-pkg.org/pkg/okcupiddata)

R package of cleaned profile data from **the 2021 revision** of [OkCupid
Profile Data for Introductory Statistics and Data Science
Courses](https://doi.org/10.1080/10691898.2015.11889737) (Journal of
Statistics Education 2015): 59,946 OkCupid users who were living within
25 miles of San Francisco, had active profiles during a period in the
2010s, and had at least one picture in their profile.

The data in this package are a “cleaned” version of the 2021 revised
data from the above paper, in that the following variables are modified
for easier use by novices:

  - In `profiles_revised`:
      - **Missing `income` values**: Previously coded as `-1`, they are
        now coded as `NA`
      - **All other missing values**: Previously coded as `""`, they are
        now coded as `NA`
      - **`offspring` and `sign`**: String instances of `"?&rsquo;"` are
        replaced with apostrophes
  - In `essay0_revised_and_shuffled`
      - **Essay responses**: Randomly row-shuffled essay data, thereby
        uncoupling individual essays from individual user profile
        information. Due to file size restrictions, only the first 140
        characters of each user’s first essay response (`essay0`: my
        self summary) are included.

**Note**:

  - The original data, publication, code, and codebook can be found
    [here](https://github.com/rudeboybert/JSE_OkCupid).
  - The original data, and hence also this cleaned data, did not include
    usernames.
  - Permission to use this data was explicitly granted by OkCupid.
  - **Revisions in 2021**:
      - Removed the exact date the data was collected.
      - Shuffled/randomized the order of the rows of the essay data,
        thereby decoupling the essay data from the profiles data.
      - Removed URL’s in the essay data that involved the following
        domains: facebook.com, instagram.com, twitter.com,
        pinterest.com, and flickr.com.
      - Added random noise to age variable in the profiles data:
        uniformly selected between {-1, 0, +1}.
      - Removed the following variables from the profiles data:
        `location`, `last_online`

## Installation

Get the released version from CRAN:

``` r
install.packages("okcupiddata")
```

Or the development version from GitHub:

``` r
# If you haven't installed devtools yet, do so:
# install.packages("devtools")
devtools::install_github("rudeboybert/okcupiddata")
```

## Load Data

To load the revised profile data, run:

``` r
library(okcupiddata)
profiles_revised
```

To load the **row-shuffled** essay data, run:

``` r
essay0_revised_and_shuffled
```


NEWS.md
# okcupiddata 0.1.0.9000

## 2021 revision of paper

The original 2015 Journal of Statistics Education publication of
"OkCupid Profile Data for Introductory Statistics and Data Science
Courses" was revised in 2021. The following changes were made:

* The essay data was
    + Randomly row shuffled, thereby uncoupling individual user profile data from individual essay
    + URL's to the following websites were removed: Facebook, Instagram, Twitter, Pinterest, Flickr
* Random noise was added to the `age` variable: uniformly selected between {-1, 0, +1}
* The `location` and `last_online` variables were removed
* The date the data were collected was recast as "a period in the 2010s"

As such

* the main `profiles` data frame has been renamed `profiles_revised` 
* the `essay0` variable in the original `profiles` data frame is now in a separate data frame `essay0_revised_and_shuffled`



# okcupiddata 0.1.0 

## Initial version of package

Modified original OkCupid profile data as follows:

* Essay responses: Due to file size restrictions, only the first 140 characters
of each user's first essay response (`essay0`: my self summary) aree included
* Missing `income` values: Previously coded as `-1`, they are now coded as `NA`
* All other missing values: Previously coded as `""`, they are now coded as `NA`
* `offspring` and `sign`: String instances of `"?&rsquo;"` are replaced with
apostrophes
* `last_online`: Date/time strings are converted to `USA/Pacific` timezone
POSIXct date-time objects


R/okcupiddata.R
#' okcupiddata:
#'
#' R package of cleaned profile data from the 2021 revision of "OkCupid Profile Data for Introductory
#' Statistics and Data Science Courses": 59,946 OkCupid
#' users who were living within 25 miles of San Francisco, had active profiles
#' during a period in the 2010s, and had at least one
#' picture in their profile. The original data, publication, code, and codebook
#' can be found at \url{https://github.com/rudeboybert/JSE_OkCupid}.
#'
#' @docType package
#' @examples
#' library(okcupiddata)
#' data(profiles)
#' @name okcupiddata
#' @references Albert Y. Kim, Adriana Escobedo-Land (2015, revised in 2021). OkCupid Profile Data for
#' Introductory Statistics and Data Science Courses. Journal of Statistics
#' Education, 23(2). \url{https://doi.org/10.1080/10691898.2015.11889737}.
NULL
#> NULL


data-raw/process_data_sets.R
library(dplyr)
library(readr)
library(stringr)
library(lubridate)
library(here)
library(fs)

# Function to clean essay data
clean_html <- function(string){
  string <- string %>%
    # Remove HTML tags
    str_replace_all("<i>|</i>|<b>|</b>|<u>|</u>|<small>|</small>|<sup>|</sup>", " ") %>%
    str_replace_all("<em>|</em>|<strong>|</strong>|<li>|</li>|<s>|</s>", " ") %>%
    str_replace_all("<p>|</p>|<del>|</del>|<ul>|</ul>|<span>|</span>", " ") %>%
    str_replace_all("<pre>|</pre>|<ol>|</ol><ins>|</ins>|<sub>|</sub>", " ") %>%
    str_replace_all("<br />|\n", " ") %>%
    # Remove hyperlink HTML code
    gsub("<a href=.*?>", "", .) %>%
    gsub("<a class=.*?>", "", .) %>%
    str_replace_all("</a>", "") %>%
    gsub("<b class=.*?>", "", .) %>%
    # Remove URL's
    gsub("http\\S+\\s*", "", .) %>%
    # Recode miscoded symbols
    str_replace_all("&amp;", "&") %>%
    str_replace_all("&nbsp;", " ")
  return(string)
}



# Cleaned Profiles -----------------------------------------------------
# Load CSV file from GitHub repo for JSE paper.
if(!file.exists("profiles_revised.csv")){
  url <- "https://github.com/rudeboybert/JSE_OkCupid/blob/master/profiles_revised.csv.zip?raw=true"
  temp_file <- tempfile()
  download.file(url, temp_file, mode="wb")
  unzip(temp_file, "profiles_revised.csv")
  file_move("profiles_revised.csv", "data-raw/profiles_revised.csv")
}

profiles_revised <-
  here("data-raw/profiles_revised.csv") %>%
  read_csv() %>%
  # # Clean missing values
  mutate(across(everything(), ~replace(., . == "", NA))) %>%
  # Clean other values
  mutate(
    income = ifelse(income == -1, NA, income),
    offspring = str_replace_all(offspring, "&rsquo;", "'"),
    sign = str_replace_all(sign, "&rsquo;", "'")
  )
usethis::use_data(profiles_revised, overwrite = TRUE, compress = "bzip2")



# Cleaned Shuffled Essays ----------------------------------------------
# Load CSV file from GitHub repo for JSE paper.
if(!file.exists("essays_revised_and_shuffled.csv")){
  url <- "https://github.com/rudeboybert/JSE_OkCupid/blob/master/essays_revised_and_shuffled.csv.zip?raw=true"
  temp_file <- tempfile()
  download.file(url, temp_file, mode="wb")
  unzip(temp_file, "essays_revised_and_shuffled.csv")
  file_move("essays_revised_and_shuffled.csv", "data-raw/essays_revised_and_shuffled.csv")
}

essay0_revised_and_shuffled <-
  here("data-raw/essays_revised_and_shuffled.csv") %>%
  read_csv() %>%
  select(essay0) %>%
  mutate(
    essay0 = clean_html(essay0),
    essay0 = str_sub(essay0, 1, 140)
  )
usethis::use_data(essay0_revised_and_shuffled, overwrite = TRUE, compress = "bzip2")




R/data.R
#' Cleaned OkCupid profile data
#'
#' Cleaned profile data of 59,946 OkCupid users who were living within 25
#' miles of San Francisco, had active profiles during a period in the
#' 2010s, and had at least one picture in their profile. The original
#' data and codebook can be found at
#' \url{https://github.com/rudeboybert/JSE_OkCupid}.
#'
#' @details The differences between the cleaned and original version of profiles data are:
#' \describe{
#' \item{Essay Responses}{Due to file size restrictions, only the first 140
#' characters of each user's first essay response (my self summary) is included}
#' \item{Missing \code{income} values}{Previously coded as \code{-1}, they are now
#' coded as \code{NA}}
#' \item{All other missing values}{Previously coded as \code{""}, they are now coded
#' as \code{NA}}
#' \item{\code{offspring} and \code{sign}}{String instances of \code{"?&rsquo;"} are
#' replaced with apostrophes}
#' }
#' @format A \code{data.frame} with 59946 rows and 22 variables:
#' \describe{
#'   \item{age}{Age}
#'   \item{body_type}{Body type}
#'   \item{diet}{Dietary habits}
#'   \item{drinks}{Drinking habits}
#'   \item{drugs}{Drug usage habits}
#'   \item{education}{Education level}
#'   \item{ethnicity}{Ethnicity}
#'   \item{height}{Height in inches with random noise added (random uniform from {-1, 0, 1})}
#'   \item{income}{Income}
#'   \item{job}{Job}
#'   \item{offspring}{Number of offspring}
#'   \item{orientation}{Sexual orientation}
#'   \item{pets}{Number of pets}
#'   \item{religion}{Religious affiliation}
#'   \item{sex}{Sex. Note at the time OkCupid only allowed for male/female binary. This has since been relaxed.}
#'   \item{sign}{Astrological sign}
#'   \item{smokes}{Smoking habits}
#'   \item{speaks}{Languages spoken}
#'   \item{status}{Relationship status}
#' }
#' @source \url{https://github.com/rudeboybert/JSE_OkCupid}
#' @examples
#' library(tibble)
#' profiles_revised
#'
#' # If using RStudio:
#' # View(profiles)
"profiles_revised"


#' Cleaned (shuffled) OkCupid essay data
#'
#' Cleaned randomly row-shuffled essay data of 59,946 OkCupid users. To
#' unpair essays from users, the rows of the essay data were randomly
#' scrambled. Furthermore, due to R package size restrictions, only the
#' first 140 characters of each user's first essay response (my self
#' summary) is included. The original data and codebook can be found at
#' \url{https://github.com/rudeboybert/JSE_OkCupid}.
#'
#' @format A \code{data.frame} with 59946 rows and 1 variable:
#' \describe{
#'   \item{essay0}{First 140 characters of first essay response (my self summary)}
#' }
#' @source \url{https://github.com/rudeboybert/JSE_OkCupid}
#' @examples
#' library(tibble)
#' essay0_revised_and_shuffled
#'
#' # If using RStudio:
#' # View(essay0_revised_and_shuffled)
"essay0_revised_and_shuffled"
